src/mscorlib/corefx/System/IO/PathHelper.Windows.cs

```diff
@@ -216,19 +213,19 @@ private static void GetFullPathName(string path, StringBuffer fullPath)
                     throw Win32Marshal.GetExceptionForWin32Error(errorCode, path);
                 }
 
-                fullPath.Length = result;
+                fullPath.Length = checked((int)result);
             }
         }
 
-        private static uint GetInputBuffer(StringBuffer content, bool isDosUnc, out StringBuffer buffer)
+        private static int GetInputBuffer(ref StringBuffer content, bool isDosUnc, ref StringBuffer buffer)
         {
-            uint length = content.Length;
+            int length = content.Length;
 
             length += isDosUnc
-                ? (uint)PathInternal.UncExtendedPrefixLength - PathInternal.UncPrefixLength
+                ? PathInternal.UncExtendedPrefixLength - PathInternal.UncPrefixLength
                 : PathInternal.DevicePrefixLength;
 
-            buffer = new StringBuffer(length);
+            buffer.EnsureCapacity(length + 1);
```

Comment:
The capacity includes the null terminator. `+1` is for the null terminator. It should been there before as well, I think.